WordPress/src/main/java/org/wordpress/android/networking/ConnectionChangeReceiver.java

```diff
@@ -0,0 +1,41 @@
+package org.wordpress.android.networking;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+import org.wordpress.android.util.NetworkUtils;
+
+import de.greenrobot.event.EventBus;
+
+/*
+ * global network connection change receiver - declared in the manifest to monitor
+ * android.net.conn.CONNECTIVITY_CHANGE
+ */
+public class ConnectionChangeReceiver extends BroadcastReceiver {
+
+    private boolean mWasConnected = true;
```

Comment:
Testing on HTC One with 5.1 I'm finding that a new ConnectionChangeReceiver is instantiated for each onReceive invocation which means the value of this is always true when onReceive is called. Can be resolved by making it static.